src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+                {
+                    {
+                        MediaTypeNames.Application.Json,
+                        CreateResponseContainer(responseObjectSchemaName)
+                    }
+                };
+                response.Content = contentDictionary;
+            }
+
+            return response;
+        }
+
+        /// <summary>
+        /// Creates the OpenAPI description of the response payload, excluding the result records:
+        /// {
+        ///     "value": [
+        ///         {
+        ///             "resultProperty": resultPropertyValue
+        ///         }
+        ///     ]
+        /// }
+        /// </summary>
+        /// <param name="responseObjectSchemaName">Schema name of response payload.</param>
+        /// <returns>The base response object container.</returns>
+        private static OpenApiMediaType CreateResponseContainer(string responseObjectSchemaName)
+        {
+            // schema for the response's collection of result records
+            OpenApiSchema resultCollectionSchema = new()
+            {
+                Reference = new OpenApiReference()
+                {
+                    Type = ReferenceType.Schema,
+                    Id = $"{responseObjectSchemaName}"
+                }
+            };
+
+            // Schema for the response's root property "value"
+            OpenApiSchema responseRootSchema = new()
+            {
+                Type = RESPONSE_ARRAY_PROPERTY,
+                Items = resultCollectionSchema
+            };
+
+            Dictionary<string, OpenApiSchema> responseBodyProperties = new()
+            {
+                {
+                    RESPONSE_VALUE_PROPERTY,
+                    responseRootSchema
+                }
+            };
+
+            OpenApiMediaType responsePayload = new()
+            {
+                Schema = new()
+                {
+                    Properties = responseBodyProperties
+                }
+            };
+
+            return responsePayload;
+        }
+
+        /// <summary>
+        /// Builds the schema objects for all entities present in the runtime configuration.
+        /// Two schemas per entity are created:
+        /// 1) {EntityName}      -> Primary keys present in schema, used for request bodies (excluding GET) and all response bodies.
+        /// 2) {EntityName}_NoPK -> No primary keys present in schema, used for POST requests where PK is autogenerated.
+        /// Schema objects can be referenced elsewhere in the OpenAPI document with the intent to reduce document verbosity.
+        /// </summary>
+        /// <returns>Collection of schemas for entities defined in the runtime configuration.</returns>
+        private Dictionary<string, OpenApiSchema> CreateComponentSchemas()
+        {
+            Dictionary<string, OpenApiSchema> schemas = new();
+
+            foreach (string entityName in _metadataProvider.EntityToDatabaseObject.Keys.ToList())
```

Comment:
Since anyways we need the `DatabaseObject` at line 597, could avoid converting the keys into a list here and instead use KeyValuePair:
```suggestion
            foreach (KeyValuePair<string, DatabaseObject> dbo in _metadataProvider.EntityToDatabaseObject)
```

Reply:
Thanks for the suggestion, i like this better too.

Reply:
done